src/app/features/Chanels/redux/actions.js

```diff
@@ -1,24 +1,35 @@
-import { GET_CHANNELS, GET_POSTS } from './actionsTypes';
-import { requestChannels, requestPosts } from '../service';
+import { GET_CHANNELS, GET_POSTS, ADD_CHANNEL } from './actionsTypes';
+import { requestChannel, requestChannels, requestPosts } from '../service';
+import { showErrorAlert, showSuccessAlert } from '../../../redux/alertAction';
 
-export const getChannels = (channels) => ({
+export const addChannelAction = (channelName) => ({
+  type: ADD_CHANNEL,
+  payload: channelName,
+});
+
+export const getChannelsAction = (channels) => ({
   type: GET_CHANNELS,
   payload: channels,
 });
 
-export const getPosts = (posts) => ({
+export const getPostsAction = (posts) => ({
   type: GET_POSTS,
   payload: posts,
 });
 
 export function fetchChannels() {
   return async (dispatch, getState) => {
     const channelsExists = getState().channelsState.length > 0;
-
     if (!channelsExists) {
-      requestChannels().then((result) => {
-        dispatch(getChannels(result));
-      });
+      const channels = localStorage.getItem('channelsList');
```

Comment:
Please, encapslate localStorage logic in **requestChannels()** function.

The end result should be something like this:
```
if (!channelsExists) {
    const channels = await requestChannels();
    dispatch(getChannelsAction(channels))
}
```